### PR TITLE
WithProductionOverlay: replace ITick => INotifyCreated & add INotifyOwnerChanged

### PR DESCRIPTION
**Pull request by atlimit8**
Allows WithProductionOverlay to update which `ProductionQueue` is associated with it, including after owner changes, based on the first enabled `ProductionQueue`.

Replaces the use of `ITick.Tick(Actor self)` with `INotifyCreated.Created(Actor self)` and `INotifyBuildComplete.BuildingComplete(Actor self)` for executing initialization code.

### CODE REVIEW

File: OpenRA.Mods.Common/Traits/Render/WithProductionOverlay.cs [hunk 1]
```diff
@@ -50,7 +51,8 @@ public WithProductionOverlay(Actor self, WithProductionOverlayInfo info)
 			var rs = self.Trait<RenderSprites>();
 			var body = self.Trait<IBodyOrientation>();
 
-			buildComplete = !self.HasTrait<Building>(); // always render instantly for units
+			buildComplete = !self.Info.Traits.Contains<BuildingInfo>(); // always render instantly for units
+			production = self.Info.Traits.Get<ProductionInfo>();
```
Comment (Mailaender) [inline OpenRA.Mods.Common/Traits/Render/WithProductionOverlay.cs]: Needs a `Requires<ProductionInfo>` then I guess.

Comment (atlimit8) [inline OpenRA.Mods.Common/Traits/Render/WithProductionOverlay.cs]: Added.

**Comment by Unit158** [comment]
Why do we want this?

**Comment by atlimit8** [comment]
The watched `ProductionQueue` changes on owner change based on first enabled `ProductionQueue` after owner change and replacing the use of `ITick` for initialization is rather obvious.

**Comment by atlimit8** [comment]
Updated description above.

**Comment by Mailaender** [comment]
Can't spot regressions in d2k. :white_check_mark:

**Comment by RoosterDragon** [comment]
:+1: Looks good to me.

**Pull request merged**